src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,247 @@
+        """ Get a dictionary of imported module version numbers
+
+        Use a variaty of method, for example a module.version call, to attempt to get the module version of each
+        module that has been imported in this instance of running SasView. The sys.modules command lists the
+        imported modules. A list of modules whose version number cannot be found is also included.
+
+        Returns
+        -------
+        module_versions_dict : dict
+            A dictionary with the module names as the key, with their respective version numbers as the value.
+        """
+        module_versions_dict = {'python': sys.version, 'SasView': sas.sasview.__version__}
+        unattainable_modules = []
+        # Generate a list of standard modules by looking at the local python library
+        standard_lib = [path.stem.split('.')[0] for path in pathlib.Path(pathlib.__file__).parent.absolute().glob('*')]
+
+        for mod in sys.modules.keys():
+
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if mod in sys.builtin_module_names or mod.split('.')[0] in standard_lib or mod.split('.')[0] == 'sas':
+                continue
+
+            # Modules that require specific methods to get the version number
+            if "PyQt5" in mod:
+                try:
+                    from PyQt5.QtCore import QT_VERSION_STR, PYQT_VERSION_STR
+                except:
+                    unattainable_modules.append(["Qt", "PyQt"])
+                else:
+                    module_versions_dict["Qt"] = QT_VERSION_STR
+                    module_versions_dict["PyQt"] = PYQT_VERSION_STR
+                continue
+
+            # Different attempts of retrieving the modules version
+            try:
+                module_versions_dict[mod] = __import__(mod).__version__
+            except AttributeError:
+                # Module has no __version__ attribute
+                pass
+            except Exception as x:
+                # Unable to access module
+                logging.error(f"{x} when attempting to access {mod} version using .__version__")
+                pass
+            else:
+                continue
+
+            try:
+                module_versions_dict[mod] = __import__(mod).version
+            except AttributeError:
+                # Module has no .version attribute
+                pass
+            except Exception as x:
+                # Unable to access module
+                logging.error(f"{x} when attempting to access {mod} version using .version")
+                pass
+            else:
+                continue
+
+            # Unreliable, so last option
+            try:
+                module_versions_dict[mod] = pkg_resources.get_distribution(mod).version
+            except:
+                # Modules that cannot be found by pkg_resources
+                pass
+            else:
+                continue
```

Comment:
Are you saying that pkg_resources fails in the distributed app?

Reply:
I haven't checked recently, but I believe pyinstaller creates a zip file with the package tree and doesn't use site packages and the associated egg info for the installed application. I don't know for sure that pkg_resource is using egg info but it seems likely. Without the egg info, the pkg_resources may trigger an exception or return None so that the `.version` reference triggers an exception. This will cause a fall-through to the "package version not found" condition.

So not a crash just a useless result. Adding two menu items to the user interface that do not work in the shipped application does not feel like a good design choice, even if it doesn't cause a crash.